test/parallel/test-module-hooks-preload.js

```diff
@@ -0,0 +1,49 @@
+'use strict';
+
+require('../common');
+const fixtures = require('../common/fixtures.js');
+const { spawnSyncAndAssert } = require('../common/child_process.js');
+
+spawnSyncAndAssert(process.execPath,
```

Comment:
Why do these all need to be sync and run serially? Can’t we use `spawnPromisified` to run them all in parallel?

Reply:
`spawnPromisified` provides very poor output when the test fails, `spawnSyncAndAssert` provides a lot more details e.g. when it fails it logs the commands and the environment variables, and both the stderr and stdout. If someone ports an async alternative that provides as much information as `spawnSyncAndAssert` I could consider using them, until then, I find `spawnPromisified` to be very frustrating when the test fails.

For child_process.spawn on a very fast process, you don't necessarily get a lot out of parallelism, considering the overhead of async handling and output buffering.